In [1]:
import pandas as pd

In [2]:
df = pd.read_csv(r'african_crises.csv')

In [4]:
df.head(10)

case  cc3  country  year  systemic_crisis  exch_usd  \
0     1  DZA  Algeria  1870                1  0.052264   
1     1  DZA  Algeria  1871                0  0.052798   
2     1  DZA  Algeria  1872                0  0.052274   
3     1  DZA  Algeria  1873                0  0.051680   
4     1  DZA  Algeria  1874                0  0.051308   
5     1  DZA  Algeria  1875                0  0.051546   
6     1  DZA  Algeria  1876                0  0.051867   
7     1  DZA  Algeria  1877                0  0.051867   
8     1  DZA  Algeria  1878                0  0.051948   
9     1  DZA  Algeria  1879                0  0.052029   

   domestic_debt_in_default  sovereign_external_debt_default  \
0                         0                                0   
1                         0                                0   
2                         0                                0   
3                         0                                0   
4                         0                                0   
5                         0                                0   
6                         0                                0   
7                         0                                0   
8                         0                                0   
9                         0                                0   

   gdp_weighted_default  inflation_annual_cpi  independence  currency_crises  \
0                   0.0              3.441456             0                0   
1                   0.0             14.149140             0                0   
2                   0.0             -3.718593             0                0   
3                   0.0             11.203897             0                0   
4                   0.0             -3.848561             0                0   
5                   0.0            -20.924178             0                0   
6                   0.0             -1.769547             0                0   
7                   0.0             29.116045             0                0   
8                   0.0             -1.492537             0                0   
9                   0.0            -16.831357             0                0   

   inflation_crises banking_crisis  
0                 0         crisis  
1                 0      no_crisis  
2                 0      no_crisis  
3                 0      no_crisis  
4                 0      no_crisis  
5                 0      no_crisis  
6                 0      no_crisis  
7                 1      no_crisis  
8                 0      no_crisis  
9                 0      no_crisis

In [5]:
df.tail(10)

case  cc3   country  year  systemic_crisis      exch_usd  \
1049    70  ZWE  Zimbabwe  2004                1  5.600000e-23   
1050    70  ZWE  Zimbabwe  2005                1  8.460000e-22   
1051    70  ZWE  Zimbabwe  2006                1  3.000000e-19   
1052    70  ZWE  Zimbabwe  2007                1  1.900000e-16   
1053    70  ZWE  Zimbabwe  2008                1  2.000000e-03   
1054    70  ZWE  Zimbabwe  2009                1  3.548000e+02   
1055    70  ZWE  Zimbabwe  2010                0  3.782000e+02   
1056    70  ZWE  Zimbabwe  2011                0  3.619000e+02   
1057    70  ZWE  Zimbabwe  2012                0  3.619000e+02   
1058    70  ZWE  Zimbabwe  2013                0  3.619000e+02   

      domestic_debt_in_default  sovereign_external_debt_default  \
1049                         1                                1   
1050                         1                                1   
1051                         1                                1   
1052                         1                                1   
1053                         1                                1   
1054                         1                                1   
1055                         1                                1   
1056                         1                                1   
1057                         1                                1   
1058                         1                                1   

      gdp_weighted_default  inflation_annual_cpi  independence  \
1049                   0.0          1.327468e+02             1   
1050                   0.0          5.858444e+02             1   
1051                   0.0          1.281114e+03             1   
1052                   0.0          6.627989e+04             1   
1053                   0.0          2.198970e+07             1   
1054                   0.0         -7.670000e+00             1   
1055                   0.0          3.217000e+00             1   
1056                   0.0          4.920000e+00             1   
1057                   0.0          3.720000e+00             1   
1058                   0.0          1.632000e+00             1   

      currency_crises  inflation_crises banking_crisis  
1049                1                 1         crisis  
1050                1                 1         crisis  
1051                1                 1         crisis  
1052                1                 1         crisis  
1053                1                 1         crisis  
1054                1                 0         crisis  
1055                0                 0      no_crisis  
1056                0                 0      no_crisis  
1057                0                 0      no_crisis  
1058                0                 0      no_crisis

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1059 entries, 0 to 1058
Data columns (total 14 columns):
case                               1059 non-null int64
cc3                                1059 non-null object
country                            1059 non-null object
year                               1059 non-null int64
systemic_crisis                    1059 non-null int64
exch_usd                           1059 non-null float64
domestic_debt_in_default           1059 non-null int64
sovereign_external_debt_default    1059 non-null int64
gdp_weighted_default               1059 non-null float64
inflation_annual_cpi               1059 non-null float64
independence                       1059 non-null int64
currency_crises                    1059 non-null int64
inflation_crises                   1059 non-null int64
banking_crisis                     1059 non-null object
dtypes: float64(3), int64(8), object(3)
memory usage: 115.9+ KB


In [8]:
df.shape

(1059, 14)

In [9]:
df.country.unique()

array(['Algeria', 'Angola', 'Central African Republic', 'Ivory Coast',
       'Egypt', 'Kenya', 'Mauritius', 'Morocco', 'Nigeria',
       'South Africa', 'Tunisia', 'Zambia', 'Zimbabwe'], dtype=object)

In [11]:
df.dtypes

case                                 int64
cc3                                 object
country                             object
year                                 int64
systemic_crisis                      int64
exch_usd                           float64
domestic_debt_in_default             int64
sovereign_external_debt_default      int64
gdp_weighted_default               float64
inflation_annual_cpi               float64
independence                         int64
currency_crises                      int64
inflation_crises                     int64
banking_crisis                      object
dtype: object

sort countries with highest systemic_crisis

In [17]:
#view only countries with highest systemic_crisis

sc = df[["country","systemic_crisis"]]
sc

country  systemic_crisis
0      Algeria                1
1      Algeria                0
2      Algeria                0
3      Algeria                0
4      Algeria                0
5      Algeria                0
6      Algeria                0
7      Algeria                0
8      Algeria                0
9      Algeria                0
10     Algeria                0
11     Algeria                0
12     Algeria                0
13     Algeria                0
14     Algeria                0
15     Algeria                0
16     Algeria                0
17     Algeria                0
18     Algeria                0
19     Algeria                0
20     Algeria                0
21     Algeria                0
22     Algeria                0
23     Algeria                0
24     Algeria                0
25     Algeria                0
26     Algeria                0
27     Algeria                0
28     Algeria                0
29     Algeria                0
...        ...              ...
1029  Zimbabwe                0
1030  Zimbabwe                0
1031  Zimbabwe                0
1032  Zimbabwe                0
1033  Zimbabwe                0
1034  Zimbabwe                0
1035  Zimbabwe                0
1036  Zimbabwe                0
1037  Zimbabwe                0
1038  Zimbabwe                0
1039  Zimbabwe                0
1040  Zimbabwe                1
1041  Zimbabwe                1
1042  Zimbabwe                1
1043  Zimbabwe                1
1044  Zimbabwe                1
1045  Zimbabwe                1
1046  Zimbabwe                1
1047  Zimbabwe                1
1048  Zimbabwe                1
1049  Zimbabwe                1
1050  Zimbabwe                1
1051  Zimbabwe                1
1052  Zimbabwe                1
1053  Zimbabwe                1
1054  Zimbabwe                1
1055  Zimbabwe                0
1056  Zimbabwe                0
1057  Zimbabwe                0
1058  Zimbabwe                0

[1059 rows x 2 columns]

In [27]:
sc.groupby(["country"]).sum().sort_values(by='systemic_crisis', ascending=False)

systemic_crisis
country                                  
Central African Republic               19
Zimbabwe                               15
Kenya                                  13
Nigeria                                10
Egypt                                   6
Tunisia                                 5
Algeria                                 4
Ivory Coast                             4
Zambia                                  4
Morocco                                 2
Angola                                  0
Mauritius                               0
South Africa                            0

sort countries with lowest sovereign_external_debt_default

In [28]:
df[["country","sovereign_external_debt_default"]].groupby(["country"]).sum().sort_values(by='sovereign_external_debt_default')

sovereign_external_debt_default
country                                                  
Mauritius                                               0
South Africa                                            5
Algeria                                                 6
Egypt                                                   6
Kenya                                                   6
Morocco                                                 6
Tunisia                                                 7
Nigeria                                                 9
Zambia                                                 12
Angola                                                 13
Ivory Coast                                            29
Zimbabwe                                               30
Central African Republic                               33

Top 5 countries with highest currency_crises and inflation_crises

In [45]:
df[["country", "currency_crises","inflation_crises"]].groupby(["country"]).sum().sort_values(by ='currency_crises', ascending=False).head(5)

currency_crises  inflation_crises
country                                        
Angola                     22                26
Zimbabwe                   21                19
Zambia                     19                20
South Africa               16                 1
Nigeria                    10                12

In [53]:
res = df[["country", "inflation_annual_cpi"]]
res=res.groupby(["country"])
round(res.mean())

inflation_annual_cpi
country                                       
Algeria                                   10.0
Angola                                   152.0
Central African Republic                   6.0
Egypt                                      5.0
Ivory Coast                                6.0
Kenya                                      9.0
Mauritius                                  7.0
Morocco                                    9.0
Nigeria                                   15.0
South Africa                               5.0
Tunisia                                    9.0
Zambia                                    22.0
Zimbabwe                              245106.0

Conclusion: From year 1870 to 2018 worst affected country is Zimbabwe 
where inflation has grown by 245106%